pkg/inngest/inngest/_internal/step_lib/step_async.py

```diff
@@ -212,7 +212,25 @@ async def run(
             except Exception as err:
                 transforms.remove_first_traceback_frame(err)
 
-                step_info.op = server_lib.Opcode.STEP_ERROR
+                max_attempts = self._execution._request.ctx.max_attempts
+                if (
+                    max_attempts is not None
+                    and self._execution._request.ctx.attempt + 1 >= max_attempts
+                ):
+                    step_info.op = server_lib.Opcode.STEP_FAILED
+                    step_failed_error = errors.StepFailedError(
+                        message=str(err),
+                        name=type(err).__name__,
+                        stack=transforms.get_traceback(err),
+                    )
```

Comment:
What's the purpose of overriding the error? Is it to prevent step retries? If so, isn't that already implicit in the `StepFailed` opcode?